receipt/src/main/java/com/hyperwallet/android/ui/receipt/view/ReceiptDetailFragment.java

```diff
@@ -0,0 +1,153 @@
+ */
+package com.hyperwallet.android.ui.receipt.view;
+
+import static com.hyperwallet.android.ui.receipt.view.ReceiptViewUtil.AMOUNT_FORMAT;
+import static com.hyperwallet.android.ui.receipt.view.ReceiptViewUtil.DETAIL_DATE_TIME_12H_FORMAT;
+import static com.hyperwallet.android.ui.receipt.view.ReceiptViewUtil.DETAIL_DATE_TIME_24H_FORMAT;
+
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.text.format.DateFormat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.IdRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.hyperwallet.android.model.receipt.Receipt;
+import com.hyperwallet.android.model.receipt.ReceiptDetails;
+import com.hyperwallet.android.ui.common.util.DateUtils;
+import com.hyperwallet.android.ui.receipt.R;
+import com.hyperwallet.android.ui.receipt.viewmodel.ReceiptDetailViewModel;
+
+import java.text.DecimalFormat;
+
+public class ReceiptDetailFragment extends Fragment {
+
+    private ReceiptDetailViewModel mReceiptDetailViewModel;
+
+    public ReceiptDetailFragment() {
+        setRetainInstance(true);
+    }
+
+    public static ReceiptDetailFragment newInstance() {
+        ReceiptDetailFragment fragment = new ReceiptDetailFragment();
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mReceiptDetailViewModel = ViewModelProviders.of(requireActivity()).get(ReceiptDetailViewModel.class);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_receipt_detail, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        Receipt receipt = mReceiptDetailViewModel.getReceipt();
+
+        // transactions
+        ReceiptViewUtil util = new ReceiptViewUtil();
+        util.setTransactionView(receipt, view);
+
+        // fee details
+        setFeeDetailsView(receipt, view);
+
+        // receipt details
+        setDetailsView(receipt, view);
+    }
+
+    private void setFeeDetailsView(@NonNull final Receipt receipt, @NonNull final View view) {
+        if (!TextUtils.isEmpty(receipt.getFee())) {
+            view.findViewById(R.id.fee_details_layout).setVisibility(View.VISIBLE);
+            double feeAmount = Double.parseDouble(receipt.getFee());
+            double amount = Double.parseDouble(receipt.getAmount());
+            double transferAmount = amount - feeAmount;
+            DecimalFormat decimalFormat = new DecimalFormat(AMOUNT_FORMAT);
```

Comment:
Could you please add a "todo" here for us to get back and apply the correct formatting for all currencies we support?